### PR TITLE
chore(applications): updated azure-identity to 4.2.1 (applics-495)

### PR DESCRIPTION
**Pull request by francescobrenci-AR**
## Describe your changes

Updated azure-identity from 4.2.0 to 4.2.1

## Issue ticket number and link

https://pins-ds.atlassian.net/browse/APPLICS-495

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/functions/applications-migration/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "mysql2": "^3.9.8",
     "pins-data-model": "github:Planning-Inspectorate/data-model#1.4.1",
     "sequelize": "^6.33.0",
-    "tedious": "^18.2.0"
+    "tedious": "^11.0.8"
```
Comment (Yakitrak) [inline apps/functions/applications-migration/package.json:26]: Is it intentional we are downgrading?

Comment (francescobrenci-AR) [reply apps/functions/applications-migration/package.json:26]: yes, looks like `tedious >=11.0.9` uses an old version of `azure-identity` (3.4.1) so to fix a vulnerability with it it downgrades it. Will look into it, Pr set to `DRAFT` for now

Comment (francescobrenci-AR) [reply apps/functions/applications-migration/package.json:26]: I discussed this with @JamesP-KC and `tedious` is an important dependency to access ODW and downgrading it to such an earlier version will most likely carry breaking changes.

The vulnerability that `azure-identity <4.2.1` has is classed as `medium` and seems low risk for auth code that isn't public facing.

There is an issue raised on the `tedious` repo about this, so hopefully a new version will not be too far away, which uses the new version of azure-identity: https://github.com/tediousjs/tedious/issues/1633.

So the solution for now would be to upgrade all the instances of `azure-identity` to 4.2.1 on our `package.json` files but leave `tedious` to 18 even if it depends on `azure-identity 3.4.1`.

@beejjacobs, @JamesP-KC , @Yakitrak does this sound good?

Comment (Yakitrak) [reply apps/functions/applications-migration/package.json:26]: That sounds like a good idea to me

**Review by Yakitrak — APPROVED** [review]

File: apps/functions/applications-migration/package.json (hunk 1 above)
Comment (beejjacobs) [reply apps/functions/applications-migration/package.json]: Yup sounds sensible.

**Review by beejjacobs — APPROVED** [review]

**Pull request merged**